### ISSUE TITLE
ftp symbolic linking / alternative endpoint generation looks to have stopped? 

### ISSUE THREAD

**Issue by Paul-Davis** [WormBase/website#7251]
Hi Webteam,

I have a question about gene names from the Uniprot team at ebi, they pick up data from the stable ftp endpoints for current_development

Example:
ftp://ftp.wormbase.org/pub/wormbase/species/c_briggsae/sequence/protein/c_briggsae.canonical_bioproject.current_development.protein.fa.gz

They are reporting old names as appearing in the file.....When I look about on the ftp://ftp.wormbase.org/pub/wormbase/species branch of the ftp it looks like the ftp deployment hasn't been working properly since ~WS260

Could someone take a look pls

Thanks

Paul

**Comment by tharris** [comment]
It likes there are a few things happening here.

First, the automatic maintenance of the FTP organization seems to be working as expected.  The paths that you mention predate the arrival of bioprojects.

We've retained those paths for two reasons: 1) they contain links to older releases that do not follow the bioproject model; 2) for users that may have linked to them.

If you want to see the file paths that are maintained programmatically, you need to look into the bioproject directories. 

The current development C. briggsae protein file, it can be found here:
ftp://ftp.wormbase.org/pub/wormbase/species/c_briggsae/PRJNA10731/sequence/protein/c_briggsae.canonical_bioproject.current_development.protein.fa.gz

And for your example,
ftp://ftp.wormbase.org/pub/wormbase/species/[SPECIES]/[BIOPROJECT]

That said, this is a good point for discussion.  The `species` directory is entirely organizational.  Should we just do away with the `SPECIES/sequence` (eg) directories altogether? Or should we remove the distinct bioproject directories so that all the symlinks for all bioprojects reside in, say, [SPECIES]/GFF?

**Comment by tharris** [comment]
(actually my answer isn't entirely correct -- still investigating...) Stay tuned for further details...

**Comment by tharris** [comment]
Okay, so I've expanded the FTP structure manager to maintain those top level directories (`species/SPECIES/gff`) and so on.  Still think we might wish to revisit the structure a little bit. It's good, but it could be still cleaner.

**Comment by Paul-Davis** [comment]
Thanks @tharris I'll take a look and then get back to Uniprot to say that it's been resolved.

**Comment by Paul-Davis** [comment]
@tharris just tried to provide updated links to Uniprot as they complained that the old "canonical_bioproject.current_development" no longer exist and cause their import pipeline to fail.

I assumed they needed to update to use the urls with the bioproject, but it looks like the old "canonical_bioproject.current_development" links are in place but for some reason they are returning 550 permission denied?

It looks like something possibly went wrong during the process as the files for WS272 aren't complete.

c_briggsae.PRJNA10731.WS272.wormpep_package.tar.gz doesn't exist.

Paul

**Comment by tharris** [comment]
The "canonical_bioproject" is a stand in for providing an actual bioproject ID.  It's used to represent what we consider the, well, canonical bioproject for a given species.

The problem earlier is that UniProt was following a path that we stopped maintaining long ago.  I've restored these links although I think they make less sense now that we are accumulating bioprojects. It looks like in doing so I missed a path.  It should be fixed now.
